### PR TITLE
[CLEANUP] remove unused and outdated wavelet code

### PR DESCRIPTION
**Pull request by timosachsenberg**
## Description

Removed some old and IMHO outdated/superseded algorithms and tools in an effort to clean up the
library.

Note: This is a conditional removal. If someone complains in the next version, we can consider pulling them (or parts of them back in).
These algorithms have been developed on machines used at least 11 years ago. They typically assume
that isotope patterns are not resolved or highly overlapping, data is sparse but recorded in profile mode, etc.

Applying them to modern TOF data fails or produces bad results. In addition, some algorithms have TODOs that suggest they are unfinished but it is unrealistic to assume that someone will pick them up again.

**removed tools:**
FeatureFinderIsotopeWavelet
PeakPickerWavelet
PrecursorMassCorrector (note: high-res version still exists)
TOFCalibration  (note: high-res version still exists)
ERPairFinder

**removed tutorials:**
Tutorial_PeakPickerCWT
Tutorial_TOFCalibration

**removed tests:**
TOFCalibration_test
ContinuousWaveletTransformNumIntegration_test
ContinuousWaveletTransform_test
FeatureFinderAlgorithmIsotopeWavelet_test
IsotopeWaveletTransform_test
IsotopeWavelet_test
OptimizePeakDeconvolution_test
OptimizePick_test
PeakPickerCWT_test
PeakShape_test

## Checklist
- [x] Make sure that you are listed in the AUTHORS file
- [x] Add relevant changes and new features to the CHANGELOG file
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Comment by jpfeuffer** [comment]
Yes. I think I never used any of those algorithms.
But can you maybe describe for others why you think each of them are outdated? Old mass spec types not common anymore, better resolution, vendor software already does corrections..?

**Comment by cbielow** [comment]
The Wavelet Code from Eva (PeakPickerWavelet) is actually quite beautiful (mathematically -- I looked over her thesis some time ago).
Keeping the code only hurts if it increases compile time (we should measure) IMHO. If something is used or not is always hard to judge IMHO, and once it's gone, the chances that someone complains instead of just going somewhere else or wasting time in locally porting it back without us knowing are high.
So i'd say, if the compile time gains are negligible, let the PP-Wavelet code stay.
I have no strong feelings about the other tools. I remember FF-Isowavelet to be slow as hell and not very good, but I could be wrong. And for TOFCalibration you need the raw tof values. Not even sure they are easy to obtain these days...

**Comment by timosachsenberg** [comment]
> The Wavelet Code from Eva (PeakPickerWavelet) is actually quite beautiful (mathematically -- I looked over her thesis some time ago). Keeping the code only hurts if it increases compile time (we should measure) IMHO.

I think this is a valid point of view, but let me share why I personally have issues with this line of argument. Compile time is not even that important to me here. It is more having stuff that no developer I know used in the last 10 years. To me this a clear sign that it should have left the library years ago. 
It also seems to be barely used (Leipzig is top 1 with the majority of calls, but I know someone who wraps OpenMS for Galaxy instances there). Even if someone uses it in production (which is not unlikely), then they will probably never update the particular OpenMS version anyway because it is so embedded into an internal process that they would never risk it. Looking at the logs, it seems that someone is using it with OpenMS 2.5 in the US.

> If something is used or not is always hard to judge IMHO, and once it's gone, the chances that someone complains instead of just going somewhere else or wasting time in locally porting it back without us knowing are high. 

I think this is a valid point but think that this can be communicated in the CHANGELOG.

File: CHANGELOG [hunk 1]
```diff
@@ -32,8 +32,36 @@ Fixes:
 Misc:
 - OpenMSInfo reports the ILP solver (CoinOr or glpk) (#7156)
 
+Note: The goal of our library is to provide useful, reusable code in a way that’s easy to understand and use.
```
Comment (timosachsenberg) [inline CHANGELOG:35]: @cbielow e.g., something like this could minimize impact on existing user bases

**Comment by timosachsenberg** [comment]
@cbielow e.g., something like this https://github.com/OpenMS/OpenMS/pull/7186/commits/4dd4e7b174a993d4259710cc4670c70a78a7e1d8 could minimize impact on our existing user base

**Review by poshul — APPROVED** [review]

**Pull request merged**